tests/components/unifiprotect/test_config_flow.py

```diff
@@ -1022,3 +1034,889 @@ async def test_discovery_can_be_ignored(hass: HomeAssistant) -> None:
+    hass: HomeAssistant,
+    bootstrap: Bootstrap,
+    nvr: NVR,
+    ufp_reconfigure_entry: MockConfigEntry,
+    mock_api_bootstrap: Mock,
+    mock_api_meta_info: Mock,
+) -> None:
+    """Test reconfigure continues even if session clearing fails."""
+    ufp_reconfigure_entry.add_to_hass(hass)
+
+    result = await ufp_reconfigure_entry.start_reconfigure_flow(hass)
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "reconfigure"
+
+    nvr.mac = _async_unifi_mac_from_hass(MAC_ADDR)
+    bootstrap.nvr = nvr
+
+    # Simulate session clear failure - should still continue
+    with (
+        patch(
+            "homeassistant.components.unifiprotect.config_flow.async_create_api_client"
+        ) as mock_create_client,
+        patch(
+            "homeassistant.config_entries.ConfigEntries.async_reload",
+        ) as mock_reload,
+    ):
+        mock_protect = AsyncMock()
+        mock_protect.clear_session = AsyncMock(side_effect=Exception("Session error"))
+        mock_create_client.return_value = mock_protect
+
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: "1.1.1.2",
+                CONF_PORT: 443,
+                CONF_VERIFY_SSL: False,
+                CONF_USERNAME: "new-username",  # Changed
+                CONF_PASSWORD: "new-password",
+                CONF_API_KEY: "new-api-key",
+            },
+        )
+        await hass.async_block_till_done()
+
+    # Should still succeed despite session clear failure
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "reconfigure_successful"
+    assert ufp_reconfigure_entry.data[CONF_USERNAME] == "new-username"
+    assert ufp_reconfigure_entry.data[CONF_PASSWORD] == "new-password"
+    mock_reload.assert_called_once()
+
+
+async def test_form_api_key_client_error(
+    hass: HomeAssistant,
+    bootstrap: Bootstrap,
+    nvr: NVR,
+    mock_api_bootstrap: Mock,
+) -> None:
+    """Test that ClientError during API key validation shows cannot_connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {}
+
+    bootstrap.nvr = nvr
+
+    with patch(
+        "homeassistant.components.unifiprotect.config_flow.ProtectApiClient.get_meta_info",
+        side_effect=ClientError("Connection failed"),
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                CONF_HOST: "1.1.1.1",
+                CONF_PORT: 443,
+                CONF_VERIFY_SSL: False,
+                CONF_USERNAME: "test-username",
+                CONF_PASSWORD: "test-password",
+                CONF_API_KEY: "test-api-key",
+            },
+        )
+
+    assert result["type"] is FlowResultType.FORM
```

Comment:
Please finish all config flow tests to reach a terminal state (ABORT or CREATE_ENTRY)